### DISCUSSION TITLE
[ROS] /goal_poseとtfについて

### DISCUSSION THREAD

**Discussion by mertcookimg** [pf-robotics/kachaka-api#72] [Q&A]
カチャカAPIのROSに関してですが、/goal_poseで指定したxy等の値とtfで確認したxyの値がずれが大きい時が時々あります。要因は色々とあるかと思いますが、一つの解決策としてパラメータでゴール到達の閾値等変更できると嬉しく思います（もしあればすみません。。。）他にも解決策がもしあればよろしくお願いいたします。

カチャカはこちらで起動しています。
`cd ~/kachaka-api/tools/ros2_bridge
./start_bridge.sh <カチャカのIPアドレス>`

**Comment by youtalk-pfr** [comment]
残念ながら現在のところ、その設定を標準アプリ、カチャカAPI双方から設定する手段がありません。
一つの代替手段としては、Nav2などのカチャカ標準ではないナビゲーションソフトウェアを使うことだと思います。そうすれば、自由に調整いただけるはずです。
https://qiita.com/terakoji-pfr/items/0f1535b45fda58edad83

cc: @kmurase-pfr @terakoji-pfr

**Comment by kmurase-pfr** [comment]
閾値を細かく調整することは現アプリ/APIではできないですが、アプリにおいて、対象の目的地に対し"目的地の編集" -> "位置の調整を省く"をONにして頂ければ、到達判定する領域を通常より広げることができるので試してみて下さい。
![Screenshot from 2024-01-10 09-36-38](https://github.com/pf-robotics/kachaka-api/assets/98864329/fd9c97b0-fe25-4703-ad48-94e7bddcfae7)

今後のAPIでパラメータによる到達判定の閾値設定を可能にすることも是非検討させて頂きます。

**Comment by youtalk-pfr** [comment]
今回は目的地指定ではなくXYYaw直接指定なので、残念ながらこのアプリ設定では効果は期待できないです...

**Comment by kmurase-pfr** [comment]
あぁなるほど、XYYaw指定か。
他に考えられるカスタマイズ手段としては、現在位置を逐次監視して目的地XYYawに十分近付いたと判断したら無理やり止めるしかないですかね...
(cancel commandの対象でないので現在位置を/goal_poseとして再度投げ直すとかしかない?)

**Comment by mertcookimg** [comment]
ご回答頂きありがとうございます！確かに自前でNav2を起動するのがひとつの解決策ですね👀
XYYaw指定でとある研究のプロトタイプを作成していた所で気づいた次第でした。
カチャカの日常使いとしては寧ろ何も問題ありませんので、またいつかそのような機能がありましたらよろしくお願いします！（とても楽しく学生さんとともに使用させて頂いてます！ありがとうございます！）
